callout-code/src/main/java/context.proto

```diff
@@ -0,0 +1,97 @@
+/*
+ * Copyright 2020 Google LLC
+ *
+ * <p>Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file
+ * except in compliance with the License. You may obtain a copy of the License at
+ *
+ * <p>https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * <p>Unless required by applicable law or agreed to in writing, software distributed under the
+ * License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+syntax = "proto3";
+
+package apigee;
+
+import "google/protobuf/any.proto";
+
+option java_package = "com.google.apigee";
+
+/* CalloutContext holds both the MessageContext and ExecutionContext objects
+ * for a java callout. */
+message Execution {
+  MessageContext messageContext = 1;
+  ExecutionContext executionContext = 2;
+}
+
+/* MessageContext holds the request, response and error Messages along with a
+ * flow variable map and a mapping of identifier strings to FlowInfos. */
+message MessageContext {
+  Message request_message = 1;
+  Message response_message = 2;
+  Message error_message = 3;
+  map<string, string> flow_variables = 4; // flow variables mapping which refers to values within the MessageContext
+  map<string, FlowInfo> flow_info_map = 5; // mapping of string identifiers to FlowInfo objects
+
+  /* Represents any object associated with a flow. */
+  message FlowInfo {
+    string identifier = 1;
+    map<string, google.protobuf.Any> map = 2;
+  }
+}
+
+/* Representation of a Message that is an abstraction of an HTTPRequestMessage
+ * or HTTPResponseMessage. Contains headers, query parameters, and message
+ * content. */
+message Message {
+  repeated string header_names = 1;
+  map<string, Headers> headers_map = 2;
+  string content = 3;
+  repeated string query_param_names = 4;
+  map<string, QueryParameters> query_param_map = 5;
+
+  /* List of header strings. */
+  message Headers {
+    repeated string header = 1;
+  }
+
+  /* List of query parameters. */
+  message QueryParameters {
+    repeated string query_parameter = 1;
+  }
+}
+
+/* Object allowing access to proxy execution context. */
+message ExecutionContext {
+  FlowType flow_type = 1; // https://docs.apigee.com/api-platform/fundamentals/what-are-flows
+  repeated Fault faults = 2;
```

Comment:
here we don't need faults and fault.  Point out the first fault in the list is the one that causes the execution to switch to error flow.

Reply:
the original comment on faults also lgtm.

     * @return all the exceptions that occurred during the flow since
     *         multiple errors are possible during a flow